pyfr/plugins/tavg.py

```diff
@@ -358,3 +363,200 @@ def __call__(self, intg):
                     self.tstart_acc = intg.tcurr
 
                 self.tout_last = intg.tcurr
+
+
+class TavgCLIPlugin(BaseCLIPlugin):
+    name = 'tavg'
+
+    @classmethod
+    def add_cli(cls, parser):
+        cmd = cls.name
+        cp = parser.add_parser(cmd, help=f'{cmd} --help')
+        sp = cp.add_subparsers(dest='sub_cmd')
+
+        # Merge command
+        ssp = sp.add_parser('merge', help=f'{cmd} merge --help')
+        ssp.set_defaults(process=cls.merge_cli)
+        ssp.add_argument('solns', nargs='*', help='Averages to merge')
+        ssp.add_argument('output', help='Output file name')
+
+    @cli_external
+    def merge_cli(self):
+        # Open all the solution files
+        self._prepare_files(self.args.solns)
+
+        # Initialise things needed for the merge
+        self._init_tavg_merge()
+
+        with h5py.File(self.args.output, 'w') as outf:
+            # Merge the averages and stds
+            self._merge_data(outf)
+
+            # Merge runtime stats
+            self._merge_runtime_stats()
+
+            # Merge the metadata
+            self._merge_meta(outf)
+
+    @staticmethod
+    def _copy_pop(x, n=0):
+        xcopy = x.copy()
+        return xcopy.pop(n), xcopy
+
+    def _eval_fun_exprs(self, avars):
+        # Prepare the substitution dictionary
+        subs = dict(zip(self.anames, avars))
+
+        # Evaluate the function and return
+        return np.array([npeval(v, subs) for v in self.fexprs])
+
+    def _init_tavg_merge(self):
+        for i, (file, stats, cfg, dt) in enumerate(self.files):
+            fields = stats.get('data', 'fields').split(',')
+            cfgsect = stats.get('tavg', 'cfg-section')
+
+            # Mapping for expression to sorted list
+            aidx = [i for i, v in enumerate(fields) if v.startswith('avg-')]
+            anames = [fields[i].removeprefix('avg-') for i in aidx]
+            amap = [i for n, i in sorted(zip(anames, aidx))]
+
+            sidx = [i for i, v in enumerate(fields) if v.startswith('std-')]
+            snames = [fields[i].removeprefix('std-') for i in sidx]
+            smap = [i for n, i in sorted(zip(snames, sidx))]
+
+            if i == 0:
+                self.stats, self.cfg = stats, cfg
+
+                self.fields = fields
+                self.cfgsect = cfgsect
+                self.region = cfg.get(cfgsect, 'region')
+                self.uuid = file['mesh_uuid']
+
+                self.idxs  = [k for k in file if re.search(r'_idxs_p\d+$', k)]
```

Comment:
Probably not worth assigning to. Instead, I'd keep a list of file names we want to write and then loop over everything in the first file and `if k not in ...` then copy it.